test/system/src/main/java/io/pravega/test/system/framework/kubernetes/K8sClient.java

```diff
@@ -666,31 +667,63 @@ public void deletePVC(String namespace, String name) {
+        AtomicInteger retriesRemaining = new AtomicInteger(retries);
+        Supplier<Boolean> shouldRetry = () -> retriesRemaining.getAndDecrement() > 0;
 
-        return Futures.loop(shouldRetry::get,
+        return Futures.loop(shouldRetry,
                 () -> Futures.delayedFuture(Duration.ofSeconds(5), executor) // wait for 5 seconds before checking for status.
-                        .thenCompose(v -> getStatusOfPodWithLabel(namespace, labelName, labelValue)) // fetch status of pods with the given label.
-                        .thenApply(podStatuses -> podStatuses.stream()
-                                // check for pods where all containers are running.
-                                .filter(podStatus -> {
-                                    if (podStatus.getContainerStatuses() == null) {
-                                        return false;
-                                    } else {
-                                        return podStatus.getContainerStatuses()
-                                                .stream()
-                                                .allMatch(st -> st.getState().getRunning() != null);
-                                    }
-                                }).count()),
-                runCount -> { // Number of pods which are running
-                    log.info("Expected running pod count of {}:{}, actual running pod count of {}:{}.", labelValue, expectedPodCount, labelValue, runCount);
-                    if (runCount == expectedPodCount) {
-                        shouldRetry.set(false);
+                        .thenCompose(v -> getPodsWithLabel(namespace, labelName, labelValue)) // fetch status of pods with the given label.
+                        .thenApply(pods -> pods),
+                pods -> {
+                    long runCount = pods.getItems().stream().map(pod -> pod.getStatus()).filter(this::isPodRunning).count();
+                    long totalCount = pods.getItems().size();
+
+                    if (runCount == totalCount && runCount == expectedPodCount) {
+                        log.info("Attempted {} retries waiting for pods.", retries - retriesRemaining.get());
+                        // Return a simplified list of the active pods (appended with their container id) to make
+                        // parsing the logs easier.
+                        List<String> names = pods.getItems()
+                                .stream()
+                                .map(pod -> pod.getMetadata().getName())
+                                .collect(Collectors.toList());
+                        log.info("Set of running pods after scale event: {}", names);
+                        // Break out of loop.
+                        retriesRemaining.set(0);
+                    } else if (retriesRemaining.get() == 0) {
+                        log.error("Retries exhausted waiting for pod(s): <{}={}>", labelName, labelValue);
```

Comment:
Nit: Can we print the expected count and the current count here in the same log line ? This will simplify debugging.